src/cdomains/apron/apronDomain.apron.ml

```diff
@@ -768,71 +815,102 @@ struct
     dprintf "%s: %a not leq %a" (name ()) pretty x pretty y
 end
 
-module D2 = DWithOps (DHetero)
-
+module type S2 =
+sig
+  module Man: Manager
+  module Tracked : Tracked
+  module Bounds : module type of Bounds (Man)
+  include module type of AOps (Tracked) (Man)
+  include Tracked
+  include SLattice with type t = Man.mt A.t
+
+  type 'a aproncomponents_t = { apr : t; priv : 'a; }
```

Comment:
It's a bit weird that this is defined here, because nothing needs it in this module. Instead it could be defined later outside of modules using two type parameters if necessary for `ApronComponents`.

Reply:
It's extracted now. You might have a look at it again.